napari/_qt/_qplugins/_qnpe2.py

```diff
@@ -0,0 +1,210 @@
+"""Plugin related functions that require Qt.
+
+Non-Qt plugin functions can be found in: `napari/plugins/_npe2.py`
+"""
+from __future__ import annotations
+
+import inspect
+from typing import (
+    TYPE_CHECKING,
+    List,
+    Optional,
+    Tuple,
+    Union,
+    cast,
+)
+
+from app_model import Action
+from app_model.types import SubmenuItem, ToggleRule
+from magicgui.type_map._magicgui import MagicFactory
+from magicgui.widgets import FunctionGui, Widget
+from npe2 import plugin_manager as pm
+from qtpy.QtWidgets import QWidget
+
+from napari._app_model import get_app
+from napari._app_model.constants import MenuGroup, MenuId
+from napari._qt._qapp_model.injection._qproviders import _provide_window
+from napari.plugins import menu_item_template
+from napari.plugins._npe2 import (
+    _get_contrib_parent_menu,
+    get_widget_contribution,
+)
+from napari.utils.events import Event
+from napari.utils.translations import trans
+from napari.viewer import Viewer
+
+if TYPE_CHECKING:
+    from npe2.manifest import PluginManifest
+    from npe2.plugin_manager import PluginName
+    from npe2.types import WidgetCreator
+
+    from napari._qt.qt_main_window import Window
+
+
+def _get_widget_viewer_param(
+    widget_callable: WidgetCreator, widget_name: str
+) -> str:
+    """Get widget parameter name for `viewer` (if any) and check type."""
+    if inspect.isclass(widget_callable) and issubclass(
+        widget_callable,
+        (QWidget, Widget),
+    ):
+        widget_param = ""
+        # Inspection can fail when adding to bundle as it thinks widget is
+        # a builtin
+        try:
+            sig = inspect.signature(widget_callable.__init__)
+        except ValueError:
+            pass
```

Comment:
See the comment above. For more details see: https://github.com/napari/napari/pull/6189

Talley wrote the original comment, @jaimergp may know more about this. But I don't know why a bundled app would cause this.

Reply:
Could You put a comment with an explanation in the except clause?

Reply:
The comment already exists above the `try` block because in my mind we are only using `try` because we have to account for the error. I have moved the comment above the except now though.